notebooks/monitoring.sh

```diff
@@ -0,0 +1,153 @@
+        # mem total from /proc/meminfo
+        mem_total=$(get_mem_info | grep MemTotal | awk 'BEGIN { FS=" " } ; { print $2 }')
+        echo "$mem_total"
+}
+
+
+function get_mem_usage() {
+        # memTotal and memAvailable
+        mem_total=$(get_mem_total)
+        mem_available=$(get_mem_available)
+
+        # usage = 100 * mem_used / mem_total
+        mem_used=$(($mem_total-$mem_available))
+        echo "$mem_used" "$mem_total" | awk '{ print 100*($1/$2) }'
+}
+
+
+function get_cpu_info() {
+        # cpu info from /proc/stat
+        cpu_info=$(cat /proc/stat | grep "cpu ")
+        echo "$cpu_info"
+}
+
+
+function get_cpu_total() {
+        # get the total cpu usage since a given time (including idle and iowait)
+        # user+nice+system+idle+iowait+irq+softirq+steal
+        cpu_total=$(get_cpu_info| awk 'BEGIN { FS=" " } ; { print $2+$3+$4+$5+$6+$7+$8+$9 }')
+        echo "$cpu_total"
+}
+
+
+function get_cpu_used() {
+        # get the cpu usage since a given time (w/o idle or iowait)
+        # user+nice+system+irq+softirq+steal
+        cpu_used=$(get_cpu_info | awk 'BEGIN { FS=" " } ; { print $2+$3+$4+$7+$8+$9 }')
+        echo "$cpu_used"
+}
+
+
+function get_cpu_usage() {
+        # get the cpu usage since a given time (w/o idle or iowait)
+        # user+nice+system+irq+softirq+steal
+        cpu_used_cur=$(get_cpu_used)
+
+        # get the total cpu usage since a given time (including idle and iowait)
+        # user+nice+system+idle+iowait+irq+softirq+steal
+        cpu_total_cur=$(get_cpu_total)
+
+        # read in previous cpu usage values
+        read -r -a cpu_prev < ${TEMP}
+        cpu_used_prev=${cpu_prev[0]}
+        cpu_total_prev=${cpu_prev[1]}
+
+        # save current values as prev values for next iteration
+        cpu_prev[0]=$cpu_used_cur
+        cpu_prev[1]=$cpu_total_cur
+        echo "${cpu_prev[@]}" > ${TEMP}
+
+        # usage = 100 * (cpu_used_cur - cpu_used_prev) / (cpu_total_cur-cpu_total_prev)
+        echo "$cpu_used_cur" "$cpu_used_prev" "$cpu_total_cur" "$cpu_total_prev" | awk 'BEGIN {FS=" "} ; { print 100*(($1-$2)/($3-$4)) }'
+
+}
+
+
+function get_date() {
+    # i.e: Fri Oct 19 15:49:30 UTC 2018
+    echo $(date)
+}
+
+
+function get_usage() {
+        echo [$(get_date)]
+        echo \* CPU usage: "$(get_cpu_usage)"%
+        echo \* Memory usage: "$(get_mem_usage)"%
+        echo \* Disk usage: $(get_disk_usage)%
+}
+
+
+function get_summary() {
+        # display header information
+        echo ==================================
+        echo =========== MONITORING ===========
+        echo ==================================
+
+        # summary info
+        echo --- General Information ---
+        # number of cores
+        echo \#CPU: $(nproc)
+        # multiply by 10^-6 to convert KB to GB
+        echo Total Memory: $(echo $(get_mem_total) 1000000 | awk '{ print $1/$2 }')G
+        echo Total Disk space: $(df -h | grep cromwell_root | awk '{ print $2}')
```

Comment:
Fine by me either way, but a trick to avoid the repeated `echo`s is to use a heredoc:
```
cat <<EOF
==================================
=========== MONITORING ============
==================================

... etc ...
EOF
```
The downside is that the closing `EOF` can't have any indentation in front of it, so the formatting can look weird inside a function.

Reply:
Ah, also this line could call `get_disk_info`